### PR TITLE
[SRVKS-1023] Add conceptual docs on the Kourier and Istio ingresses

### PR DESCRIPTION
**Pull request by rh-max**
Version(s):
`serverless-docs-1.32`+

Issue:
https://issues.redhat.com/browse/SRVKS-1023

Link to docs preview:
https://72690--ocpdocs-pr.netlify.app/openshift-serverless/latest/knative-serving/kourier-and-istio-ingresses

QE review:
- [ ] QE has approved this change.

### CODE REVIEW

**Comment by rh-max** [comment]
@ReToCode Could you PTAL? Thank you.

**Review by ReToCode — COMMENTED** [review]
LGTM

**Comment by skopacz1** [comment]
hi @rh-max before I start to review this, I notice that you are including new images in your PR. Correct me if I'm wrong, but they don't seem to be created by the CCS graphics department. If that's the case, you'll need to work with them in order to have these images recreated following the style rules/conventions that we follow for all CCS images.

[This website](https://docs.google.com/forms/d/e/1FAIpQLSdF8V0n3E-aAfRen8vZchXCxwA0iDTd0QYyASnutznfGLATvA/viewform) provides instructions on how to request diagrams be made by the graphics department. The bright side is that your images are already very well designed, so I think the process will be a lot more straightforward.

Let me know if you'd still like me to review the other content besides the images, but you can also wait until the images are completed to get a peer review.

**Comment by rh-max** [comment]
@skopacz1 Hi, yes, please proceed with the review.

File: modules/serverless-kourier-and-istio-ingresses-overview.adoc [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+See the "Integrating Service Mesh with OpenShift Serverless" section of {ServerlessProductShortName} documentation for Istio requirements and installation instructions.
+
+== Traffic configuration and routing
+
+Regardless of whether you use Kourier or Istio, the traffic for a Knative Service is configured in the `knative-serving` namespace by the `net-kourier-controller` or the `net-istio-controller` respectively.
+
+The controller reads the `KnativeService` and its child custom resources to configure the ingress solution. Both ingress solutions provide an ingress gateway pod that becomes part of the traffic path. Both ingress solutions are based on Envoy. By default, {ServerlessProductShortName} has two routes for each `KnativeService` object:
+
+* A *cluster-external route* that is forwarded by the OpenShift router, for example `myapp-namespace.example.com`.
+* A *cluster-local route* containing the cluster domain, for example `myapp.namespace.svc.cluster.local`. This domain can and should be used to call Knative services from Knative or other user workload.
```
Comment (skopacz1) [inline modules/serverless-kourier-and-istio-ingresses-overview.adoc:41]: Plural nit:
```suggestion
* A *cluster-local route* containing the cluster domain, for example `myapp.namespace.svc.cluster.local`. This domain can and should be used to call Knative services from Knative or other user workloads.
```

**Review by skopacz1 — COMMENTED** [review]
Left a few comments but otherwise looks good to me. I don't see your images used anywhere in the PR. Even if they're not used, I'm not personally sure whether the PR would be merged with images that aren't created by the graphics department. 

If you're not going to use them now, it might be better to remove them from this PR and create another PR later, when you have images created by the graphics department and are ready to implement them into the live docs. Let me know if you have any questions!

File: install/installing-knative-serving.adoc [hunk 2]
```diff
@@ -18,6 +18,12 @@ If you want to xref:../observability/tracing/serverless-tracing.adoc#serverless-
 include::modules/serverless-install-serving-web-console.adoc[leveloffset=+1]
 include::modules/serverless-install-serving-yaml.adoc[leveloffset=+1]
 
+[id="additional-resources_installing-knative-serving"]
+[role="_additional-resources"]
+== Additional resources
+
+* xref:../knative-serving/kourier-and-istio-ingresses.adoc#kourier-and-istio-ingresses[Kourier and Istio ingresses]
+
```
[the remark this answers is not in the document]
Comment (rh-max) [reply install/installing-knative-serving.adoc:26]: I think additional resources should be closer to the text they supplement than the next steps, which are about things you do after you are finished with the current text.

**Comment by mgencur** [comment]
LGTM

**Comment by rh-max** [comment]
/label merge-review-needed
To the merge reviewer: please do not merge on completion, we are waiting for the GA confirmation. Thank you.

File: knative-serving/kourier-and-istio-ingresses.adoc [hunk 3]
```diff
@@ -0,0 +1,14 @@
+:_mod-docs-content-type: ASSEMBLY
+include::_attributes/common-attributes.adoc[]
+[id="kourier-and-istio-ingresses"]
+= Kourier and Istio ingresses
+:context: kourier-and-istio-ingresses
+
+{ServerlessProductName} supports the following two ingress solutions:
```
Comment (adellape) [inline knative-serving/kourier-and-istio-ingresses.adoc:7]: Missing TOC:
```suggestion
:context: kourier-and-istio-ingresses

toc::[]

{ServerlessProductName} supports the following two ingress solutions:
```

**Review by adellape — COMMENTED** [review]
Confirmed with Max via Slack that this is ready for merge.

**Pull request merged**